dht/dht.go

```diff
@@ -141,6 +163,12 @@ func NewServer(c *ServerConfig) (s *Server, err error) {
 		}
 	}
 	s.bootstrapNodes = c.BootstrapNodes
+	if c.NodeId != "" {
+		if !c.NoSecurity {
+			return nil, ErrConflictingConfigNoSecNodeId
```

Comment:
Fixed, and further added some logic so that ID is given to the config as a hex string rather than requiring caller to make a bytestring and cast to string.